lambda/matching.ml

```diff
@@ -1997,13 +2013,7 @@ let get_args_tuple arity p rem =
   | { pat_desc = Tpat_tuple args } -> args @ rem
   | _ -> assert false
 
-let matcher_tuple arity p rem =
-  match p.pat_desc with
-  | `Any -> omegas arity @ rem
-  | `Tuple args when List.length args = arity -> args @ rem
-  | _ -> raise NoMatch
-
-let make_tuple_matching loc arity def = function
+let make_tuple_matching p loc arity def = function
```

Comment:
I am not sure about keeping both `p` and `arity` as arguments (with the invariant that
`Pattern_head.arity @@ head p` = `arity`) since `arity` is a specialization of the head description in the tuple case.

Reply:
Yes, the commit I just added to the PR stack passes a real `head` instead of `p` and derives the arity from it. Is this satisfactory?

(The state that you observe by reviewing commit-by-commit is the result of trying to proceed by small changes; the extra "polishing" steps you suggest would make the change more invasive if we did at the same time, so it is better to have them in separate commits.)

Reply:
I find the PR more internally coherent with the last commit, thanks.
I agree that there is no need to shuffle the commits.

Reply:
Perfect, thanks!